stdlib/string.ml

```diff
@@ -326,7 +326,11 @@ let edit_distance' ?(limit = Int.max_int) s (s0, len0) s1 =
     let len1 = Array.length row - 1 in
     let row_min = ref Int.max_int in
     row.(0) <- i;
-    for j = Int.max 1 (i - limit) to Int.min len1 (i + limit - 1) do
+    let jmax =
+      let jmax = Int.min len1 (i + limit - 1) in
+      if jmax < 0 then (* overflow *) len1 else jmax
```

Comment:
I have an alternative computation using `0 < i` and `0 < limit ≤ len1`  with one less comparison and one more subtraction:
```ocaml
let jmax =
   let im = i - 1 in
   if im > len1 - limit then len1 else im + limit
```

Reply:
I think I rather keep the obvious formulation. FWIW I did measure the change with the new overflow detection and there was no noticable peformance change.

Reply:
This is fine with me.